miLAG_package/milags_functions.R

```diff
@@ -0,0 +1,1047 @@
+
+  pars_local_regr <- pars
+  pars_local_regr$tangent_method <- "local.regression"
+  data_new_exp2 <- calc_lag(data = data,
+                                 method = "exponential",
+                                 pars = pars_local_regr)%>%
+    mutate(lag_calculation_method = "tangent to \nmax growth line")
+
+  pars$threshold <- biomass_incr_threshold
+  data_new_biominc <-  calc_lag(data = data,
+                                    method = "biomass increase",
+                                    pars = pars)%>%
+    mutate(lag_calculation_method = "biomass \nincrease")
+
+
+  data_all_with_lag <- data_new_max_infl %>%
+    rbind(data_new_exp) %>%
+    rbind(data_new_exp2) %>%
+    rbind(data_new_biominc) %>%
+    rbind(data_new_baranyi) %>%
+    rbind(data_new_logistic)
+
+  data_all_with_lag$lag[data_all_with_lag$lag < 0] = NA
+  return(data_all_with_lag)
+}
+
+
+
+
+#' get_def_pars
+#' Set defaults parameters used by calc_lag function
+#' @returns list of parameters
+#' @export
+get_def_pars <- function() {
+  pars <- list(model = "logistic",
+              n0_method = "first.observation",
+              tangent_method = "local.regression",
+              threshold = 10^2,
+              curve_points = 3,
+              init_gr_rate = NULL,#0.1,
+              init_lag = NULL, #3.5,
+              algorithm = "auto",#"default", "Levenberg-Marquardt",#
+              max_iter = 100)
+  return(pars)
+}
+
+
+
+#' smooth_data
+#' Smoothens growth curves data
+#' @param data a data frame with two required columns names: "time" and "biomass",and one optional column: "curve_id"
+#' This is data from may come from multiple growth curves
+#' @param smooth_kind kind used for the smooth functions, defaults to "3RS3R"
+#' @returns smoothened data
+#' @export
+smooth_data <- function(data, smooth_kind = "3RS3R") {
+  if (!("curve_id" %in% names(data))) {
+    data$curve_id <- "growth.curve"
+  }
+  data_smooth <- data  %>% filter(FALSE)
+  for (this_curve_id in unique(data$curve_id)) {
+    data_this_curve <- data %>% filter(curve_id == this_curve_id) %>%
+      mutate(biomass_smooth = smooth(biomass, kind = smooth_kind)) %>%
+      select(time, biomass = biomass_smooth, curve_id)
+    data_smooth <- rbind(data_smooth, data_this_curve)
+  }
+  return(data_smooth)
+}
+
+
+
+#' cut_the_data
+#' Smoothens growth curves data
+#' @param data a data frame with two required columns names: "time" and "biomass",and one optional column: "curve_id"
+#' This is data from may come from multiple growth curves
+#' @param max_time max. time at which we want to cut the growth curve data
+#' @returns cut data
+#' @export
+cut_the_data <- function(data, max_time) {
+  data_short <- data %>% filter(time <= max_time)
+  return(data_short)
+}
+
+
+
+#' get_theme (not exported)
+#'
+#' This function sets a ggplot theme without grid
+#' @param text_size defaults to 12
+#' @returns a ggplot theme
+#' @export

```

Comment:
it is not exported so @export should nor be there

Reply:
Removed export from milags_functions and NAMESPACE.